### PR TITLE
Issue #1313 Addon: add minimum Minishift version to execute addons

### PR DESCRIPTION
**Pull request by praveenkumar**

### CODE REVIEW

**Comment by LalatenduMohanty** [comment]
@praveenkumar This is the right issue for the PR https://github.com/minishift/minishift/issues/1313 , the issue number you have added in the commit message is already closed.

**Comment by praveenkumar** [comment]
@LalatenduMohanty Thanks, updated the commit message.

**Comment by anjannath** [comment]
also it should be called syntax and not semantic? @robin-owen

File: docs/source/using/addons.adoc [hunk 1]
```diff
@@ -27,16 +27,18 @@ The following example shows the contents of an add-on file, including the name a
...
-oc new-app -p OPENSHIFT_OAUTH_CLIENT_SECRET=#{mytoken}                                  // <6>
-oc adm policy add-scc-to-group anyuid system:authenticated                              // <7>
+oc new-app -p OPENSHIFT_OAUTH_CLIENT_SECRET=#{mytoken}                                  // <7>
+oc adm policy add-scc-to-group anyuid system:authenticated                              // <8>
 ----
 <1> (Required) Name of the add-on.
 <2> (Required) Description of the add-on.
 <3> (Optional) Comma separated list of required interpolation variables. See xref:../using/addons.adoc#addon-variable-interpolation[Variable Interpolation]
 <4> (Optional) OpenShift version required to run a specific add-on. See xref:../using/addons.adoc#addon-openshift-version-semantics[OpenShift Version Semantic]
+<5> (Optional) {project} version required to run a specific add-on. See xref:../using/addons.adoc#addon-minishift-version-semantics[{project} Version Semantic]
```
Comment (robin-owen) [inline docs/source/using/addons.adoc]: Typo: s/Semantic/Semantics

File: docs/source/using/addons.adoc [hunk 2]
```diff
@@ -76,6 +78,32 @@ If the metadata field *OpenShift-Version* is not specified in the add-on header,
 *OpenShift-Version* only supports versions in the form of <major>.<minor>.<patch>.
 ====
 
+[[addon-minishift-version-semantics]]
+== {project}-Version Semantics
```
Comment (robin-owen) [inline docs/source/using/addons.adoc]: Does the name of this field change between upstream and downstream? Using `{project}` means the content here will change depending on the context. If the field name remains the same, we should use the actual field name.

File: docs/source/using/addons.adoc [hunk 3]
```diff
@@ -76,6 +78,32 @@ If the metadata field *OpenShift-Version* is not specified in the add-on header,
...
+
+As part of the add-on metadata you can specify the {project} version which needs to be running in order to apply the add-on.
+To do so, you can specify the optional *{project}-Version* metadata field.
```
Comment (robin-owen) [inline docs/source/using/addons.adoc]: Likewise.

File: docs/source/using/addons.adoc [hunk 4]
```diff
@@ -76,6 +78,32 @@ If the metadata field *OpenShift-Version* is not specified in the add-on header,
...
+| >1.22.0              | {project} version higher than 1.22.0 needs to be running in order to apply the add-on.
+| >=1.22.0             | {project} version 1.22.0 or higher needs to be running in order to apply the add-on.
+| 1.22.0               | {project} version 1.22.0 needs to be running in order to apply the add-on.
+| >=1.21.0, <1.25.0    | {project} version should higher or equal to 1.21.0 but lower than 1.25.0.
+| >=1.22.0, +<=1.25.0+ | {project} version should higher or equal to 1.22.0 but lower or equal to 1.25.0.
+|===
+
+[NOTE]
+====
+If the metadata field *{project}-Version* is not specified in the add-on header, the add-on can be applied against any version of {project}.
```
Comment (robin-owen) [inline docs/source/using/addons.adoc]: Same here.

File: docs/source/using/addons.adoc [hunk 5]
```diff
@@ -76,6 +78,32 @@ If the metadata field *OpenShift-Version* is not specified in the add-on header,
...
+====
+
+[NOTE]
+====
+*{project}-Version* only supports versions in the form of <major>.<minor>.<patch>.
```
Comment (robin-owen) [inline docs/source/using/addons.adoc]: And here. :)

**Comment by robin-owen** [comment]
@anjannath I think "semantics" is appropriate, as this implies meaning behind the syntax.

**Comment by anjannath** [comment]
Thanks @robin-owen i was a bit confused about the word.

**Comment by amitkrout** [comment]
I created a addon file ```test-addon.addon``` and put it in ```test-addon``` directory. 
```
# Name: test-addon
# Description: Demo add-on for checking minimum Minishift version to execute addons.
# Url: https://minishift.io
# Minishift-Version: >=1.16.0, <=1.21.0

echo "Success"
```
Then i install and apply it 
```
$ ./minishift addons install /Users/amit/test-addon/
Addon 'test-addon' installed

$ ./minishift addons apply test-addon
-- Applying addon 'test-addon':Error applying the add-on: 
Add-on does not support Minishift version 1.22.0. You need to use a version <=1.21.0
```

The reporting error message does not display the range which i set in addon metadata ```Minishift-Version```

Expecting:
```
$ ./minishift addons apply test-addon
-- Applying addon 'test-addon':Error applying the add-on: 
Add-on does not support Minishift version 1.22.0. You need to use a version >=1.16.0, <=1.21.0
```

File: docs/source/using/addons.adoc (hunk 2 above)
Comment (praveenkumar) [reply docs/source/using/addons.adoc]: @robin-owen no it will be same so you want it to be hard coded?

Comment (robin-owen) [reply docs/source/using/addons.adoc]: @praveenkumar Correct -- if it remains the same, there's no use for `{project}`. :)

**Comment by praveenkumar** [comment]
@amitkrout @anjannath all the changes in place now.

File: docs/source/using/addons.adoc [hunk 6]
```diff
@@ -76,6 +78,32 @@ If the metadata field *OpenShift-Version* is not specified in the add-on header,
 *OpenShift-Version* only supports versions in the form of <major>.<minor>.<patch>.
 ====
 
+[[addon-minishift-version-semantics]]
+== Minishift-Version Semantics
+
+As part of the add-on metadata you can specify the {project} version which needs to be running in order to apply the add-on.
+To do so, you can specify the optional *{project}-Version* metadata field.
```
Comment (gbraad) [inline docs/source/using/addons.adoc]: I am not sure if `{project}` should be used here. It is not like this becomes `CDK-Version`, right?

**Comment by anjannath** [comment]
retest this please

**Pull request merged**